net-misc/tigervnc/files/tigervnc-1.12.0.initd

```diff
@@ -0,0 +1,70 @@
+#!/sbin/openrc-run
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License, v2 or later
+
+depend() {
+	need net
+}
+
+checkconfig() {
+	if [ -n "${DISPLAYS}" ]; then
+		if [ $1 = "start" ]; then
+			local user
+			for user in $DISPLAYS; do
+				if eval [ ! -f "~${user%%:*}/.vnc/passwd" ]; then
+					eerror "There are no passwords defined for user ${user%%:*}."
+					return 1
+				elif [ -e "/tmp/.X11-unix/X${user##*:}" ]; then
+					eerror "Display :${user##*:} appears to be already in use because of /tmp/.X11-unix/X${user##*:}"
+					eerror "Remove this file if there is no X server $HOSTNAME:${user##*:}"
+					return 1
+				elif [ -e "/tmp/.X${user##*:}-lock" ]; then
+					eerror "Display :${user##*:} appears to be already in use because of /tmp/.X${user##*:}-lock"
+					eerror "Remove this file if there is no X server $HOSTNAME:${user##*:}"
+					return 1
+				elif ! grep -E "^[^#]*:${user##*:}=${user%%:*}" /etc/tigervnc/vncserver.users > /dev/null 2>&1; then
+					eerror "User ${user%%:*} is not defined for display :${user##*:} in /etc/tigervnc/vncserver.users"
+					return 1
+				fi
+			done
+		fi
+		return 0
+	else
+		eerror 'Please define $DISPLAYS in /etc/conf.d/tigervnc'
+		return 1
+	fi
+}
+
+start() {
+	checkconfig start || return 1
+	ebegin "Starting TigerVNC server"
+	for user in $DISPLAYS; do
+		usrname=${user%%:*}
+		usropts=VNC_OPTS_${usrname}
+		usrdisp=${user##*:}
+		usrvar=$usropts"["$usrdisp"]"
+		[ -z ${!usrvar} ] && vnc_opts=$VNC_OPTS || vnc_opts=${!usrvar}
+
+		/usr/libexec/vncsession-start :${user##*:} $vnc_opts &>/dev/null
+	done
+	eend $?
+}
+
+stop() {
+	checkconfig stop || return 2
+	ebegin "Stopping TigerVNC server"
+	for user in $DISPLAYS; do
+		local SESSIONPID=`cat /run/vncsession-\:${user##*:}.pid`
+		local SERVERPID=`pstree -p $SESSIONPID | grep Xvnc | sed -e 's/^.*Xvnc(//
+			s/).*$//'`
+		kill $SERVERPID &>/dev/null
```

Comment:
Not ideal.  What we have in `/run/vncsession` is not what we need to kill.  Example below:
```
gandalf ~ # cat /run/vncsession-\:1.pid 
1949
gandalf ~ # pstree -p 1949
vncsession(1949)───xinit(1950)─┬─Xvnc(1961)
                               └─xfce4-session(1963)─┬─Thunar(2117)─┬─{Thunar}(2119)
```
We need to kill `Xvnc(1961)`.  Probably finding it with `pstree` is a bit lazy, and also not sure everyone has `pstree` installed, so this should be improved.

Reply:
I added psmisc to RDEPEND and left a comment about why I am doing this.